### DISCUSSION TITLE
Resolver return types do not have to match actual return types?

### DISCUSSION THREAD

**Discussion by tetchel** [MichalLytek/type-graphql#1112] [Q&A] [answered]
Hi,

I built a simple demo backend using this library, and it works. I am overall really impressed by the DX this provides.

However, one quirk I have noticed is that the return type declared by the `@Query` decorator does not have to match the function's actual return type.

Working example:
![image](https://user-images.githubusercontent.com/8888438/141513171-d1f84ff2-57c0-45b7-8ba1-f8f1f81f8891.png)

Broken example:
![image](https://user-images.githubusercontent.com/8888438/141513327-6fbe900b-2e21-41e9-9331-6339a18e4d9f.png)

There should be an error pointing out the disconnect between the declared Query return type of a `[Customers]!` and the actual return type of `number`.

Is this simply a limitation of the library, and it's up to users to make sure the types match? Or am I missing something?

Thanks for any help.

**Comment by MichalLytek** [comment, marked as answer]
> Is this simply a limitation of the library

This is the limitation of TypeScript. In ideal world it would just read `: Promise<number>` type annotation and emit `Float!` in GraphQL.
